Store fewer raw pointers in containers in rendering code

cdumez opened the pull request:
#### b4a04231c20afc7a2800c446dd3d7ece452b6022
<pre>
Store fewer raw pointers in containers in rendering code
<a href="https://bugs.webkit.org/show_bug.cgi?id=267513">https://bugs.webkit.org/show_bug.cgi?id=267513</a>

Reviewed by Brent Fulgham.

This tested as performance neutral on the benchmarks we track.

* Source/WebCore/rendering/FloatingObjects.cpp:
(WebCore::FloatingObjects::moveAllToFloatInfoMap):
* Source/WebCore/rendering/FloatingObjects.h:
* Source/WebCore/rendering/Grid.cpp:
(WebCore::Grid::gridItemArea const):
(WebCore::Grid::setGridItemArea):
* Source/WebCore/rendering/Grid.h:
* Source/WebCore/rendering/GridMasonryLayout.cpp:
(WebCore::GridMasonryLayout::updateItemOffset):
(WebCore::GridMasonryLayout::offsetForChild const):
* Source/WebCore/rendering/GridMasonryLayout.h:
* Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp:
(WebCore::GridTrackSizingAlgorithm::cacheBaselineAlignedItem):
(WebCore::GridTrackSizingAlgorithm::computeBaselineAlignmentContext):
* Source/WebCore/rendering/GridTrackSizingAlgorithm.h:
* Source/WebCore/rendering/ImageQualityController.cpp:
(WebCore::ImageQualityController::set):
* Source/WebCore/rendering/ImageQualityController.h:
* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::getBlockRareData):
(WebCore::ensureBlockRareData):
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::hasOverridingLogicalHeight const):
(WebCore::RenderBox::hasOverridingLogicalWidth const):
(WebCore::RenderBox::setOverridingLogicalHeight):
(WebCore::RenderBox::setOverridingLogicalWidth):
(WebCore::RenderBox::clearOverridingLogicalHeight):
(WebCore::RenderBox::clearOverridingLogicalWidth):
(WebCore::RenderBox::overridingLogicalWidth const):
(WebCore::RenderBox::overridingLogicalHeight const):
(WebCore::RenderBox::overridingContainingBlockContentWidth const):
(WebCore::RenderBox::overridingContainingBlockContentHeight const):
(WebCore::RenderBox::hasOverridingContainingBlockContentWidth const):
(WebCore::RenderBox::hasOverridingContainingBlockContentHeight const):
(WebCore::RenderBox::overridingContainingBlockContentLogicalWidth const):
(WebCore::RenderBox::overridingContainingBlockContentLogicalHeight const):
(WebCore::RenderBox::hasOverridingContainingBlockContentLogicalWidth const):
(WebCore::RenderBox::hasOverridingContainingBlockContentLogicalHeight const):
(WebCore::RenderBox::setOverridingContainingBlockContentLogicalWidth):
(WebCore::RenderBox::setOverridingContainingBlockContentLogicalHeight):
(WebCore::RenderBox::clearOverridingContainingBlockContentSize):
(WebCore::RenderBox::clearOverridingContainingBlockContentLogicalHeight):
(WebCore::RenderBox::overridingLogicalHeightLength const):
(WebCore::RenderBox::setOverridingLogicalHeightLength):
(WebCore::RenderBox::hasOverridingLogicalHeightLength const):
(WebCore::RenderBox::clearOverridingLogicalHeightLength):
(WebCore::RenderBox::overridingLogicalWidthLength const):
(WebCore::RenderBox::setOverridingLogicalWidthLength):
(WebCore::RenderBox::hasOverridingLogicalWidthLength const):
(WebCore::RenderBox::clearOverridingLogicalWidthLength):
* Source/WebCore/rendering/RenderBoxModelObject.cpp:
(WebCore::RenderBoxModelObject::continuation const):
(WebCore::RenderBoxModelObject::inlineContinuation const):
(WebCore::RenderBoxModelObject::forRendererAndContinuations):
(WebCore::RenderBoxModelObject::continuationChainNode const):
(WebCore::RenderBoxModelObject::insertIntoContinuationChainAfter):
(WebCore::RenderBoxModelObject::removeFromContinuationChain):
(WebCore::RenderBoxModelObject::ensureContinuationChainNode):
(WebCore::RenderBoxModelObject::firstLetterRemainingText const):
(WebCore::RenderBoxModelObject::setFirstLetterRemainingText):
(WebCore::RenderBoxModelObject::clearFirstLetterRemainingText):
* Source/WebCore/rendering/RenderFlexibleBox.cpp:
(WebCore::RenderFlexibleBox::cachedChildIntrinsicContentLogicalHeight const):
(WebCore::RenderFlexibleBox::setCachedChildIntrinsicContentLogicalHeight):
(WebCore::RenderFlexibleBox::clearCachedChildIntrinsicContentLogicalHeight):
(WebCore::RenderFlexibleBox::cacheChildMainSize):
(WebCore::RenderFlexibleBox::clearCachedMainSizeForChild):
(WebCore::RenderFlexibleBox::computeFlexBaseSizeForChild):
(WebCore::RenderFlexibleBox::maybeCacheChildMainIntrinsicSize):
* Source/WebCore/rendering/RenderFlexibleBox.h:
* Source/WebCore/rendering/RenderFragmentContainer.cpp:
(WebCore::RenderFragmentContainer::renderBoxFragmentInfo const):
(WebCore::RenderFragmentContainer::setRenderBoxFragmentInfo):
(WebCore::RenderFragmentContainer::takeRenderBoxFragmentInfo):
(WebCore::RenderFragmentContainer::removeRenderBoxFragmentInfo):
* Source/WebCore/rendering/RenderFragmentContainer.h:

Canonical link: <a href="https://commits.webkit.org/273101@main">https://commits.webkit.org/273101@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e66cd5361a3b98fc94214412767cd6fbca650e07

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/34258 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/13069 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/36252 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/36945 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/31032 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/35341 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/15455 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/10194 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/30052 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/34770 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/11066 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/30565 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/9661 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/9764 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/30611 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/38236 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/31131 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/30928 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/35845 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/9888 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/7754 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/33769 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/11680 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/10447 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4400 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/10713 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/rendering/RenderFragmentContainer.cpp:
```diff
@@ -319,26 +319,26 @@ void RenderFragmentContainer::detachFragment()
 RenderBoxFragmentInfo* RenderFragmentContainer::renderBoxFragmentInfo(const RenderBox* box) const
```
Should the `box` parameter be a Ref, too, since we assume it is never `nullptr`, perhaps as follow-up work?

cdumez replied:
We normally don't switch parameter types to smart pointers like this unless there is ownership transfer, to avoid refcounting churn. However, you're right that it is assumed to be non-null so it should probably be a C++ reference. I'll check how big of a change that is, as I suspect this is a virtual function.

brentfulgham commented on Source/WebCore/rendering/RenderFragmentContainer.cpp:
```diff
@@ -319,26 +319,26 @@ void RenderFragmentContainer::detachFragment()
+    return m_renderBoxFragmentInfo.get(*box);
 }
 
 RenderBoxFragmentInfo* RenderFragmentContainer::setRenderBoxFragmentInfo(const RenderBox* box, LayoutUnit logicalLeftInset, LayoutUnit logicalRightInset,
```
Ditto `box` (pointer -> Ref)?

brentfulgham reviewed (approved):
LGTM. r=me

cdumez committed:
Store fewer raw pointers in containers in rendering code
https://bugs.webkit.org/show_bug.cgi?id=267513

Reviewed by Brent Fulgham.

This tested as performance neutral on the benchmarks we track.

* Source/WebCore/rendering/FloatingObjects.cpp:
(WebCore::FloatingObjects::moveAllToFloatInfoMap):
* Source/WebCore/rendering/FloatingObjects.h:
* Source/WebCore/rendering/Grid.cpp:
(WebCore::Grid::gridItemArea const):
(WebCore::Grid::setGridItemArea):
* Source/WebCore/rendering/Grid.h:
* Source/WebCore/rendering/GridMasonryLayout.cpp:
(WebCore::GridMasonryLayout::updateItemOffset):
(WebCore::GridMasonryLayout::offsetForChild const):
* Source/WebCore/rendering/GridMasonryLayout.h:
* Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp:
(WebCore::GridTrackSizingAlgorithm::cacheBaselineAlignedItem):
(WebCore::GridTrackSizingAlgorithm::computeBaselineAlignmentContext):
* Source/WebCore/rendering/GridTrackSizingAlgorithm.h:
* Source/WebCore/rendering/ImageQualityController.cpp:
(WebCore::ImageQualityController::set):
* Source/WebCore/rendering/ImageQualityController.h:
* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::getBlockRareData):
(WebCore::ensureBlockRareData):
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::hasOverridingLogicalHeight const):
(WebCore::RenderBox::hasOverridingLogicalWidth const):
(WebCore::RenderBox::setOverridingLogicalHeight):
(WebCore::RenderBox::setOverridingLogicalWidth):
(WebCore::RenderBox::clearOverridingLogicalHeight):
(WebCore::RenderBox::clearOverridingLogicalWidth):
(WebCore::RenderBox::overridingLogicalWidth const):
(WebCore::RenderBox::overridingLogicalHeight const):
(WebCore::RenderBox::overridingContainingBlockContentWidth const):
(WebCore::RenderBox::overridingContainingBlockContentHeight const):
(WebCore::RenderBox::hasOverridingContainingBlockContentWidth const):
(WebCore::RenderBox::hasOverridingContainingBlockContentHeight const):
(WebCore::RenderBox::overridingContainingBlockContentLogicalWidth const):
(WebCore::RenderBox::overridingContainingBlockContentLogicalHeight const):
(WebCore::RenderBox::hasOverridingContainingBlockContentLogicalWidth const):
(WebCore::RenderBox::hasOverridingContainingBlockContentLogicalHeight const):
(WebCore::RenderBox::setOverridingContainingBlockContentLogicalWidth):
(WebCore::RenderBox::setOverridingContainingBlockContentLogicalHeight):
(WebCore::RenderBox::clearOverridingContainingBlockContentSize):
(WebCore::RenderBox::clearOverridingContainingBlockContentLogicalHeight):
(WebCore::RenderBox::overridingLogicalHeightLength const):
(WebCore::RenderBox::setOverridingLogicalHeightLength):
(WebCore::RenderBox::hasOverridingLogicalHeightLength const):
(WebCore::RenderBox::clearOverridingLogicalHeightLength):
(WebCore::RenderBox::overridingLogicalWidthLength const):
(WebCore::RenderBox::setOverridingLogicalWidthLength):
(WebCore::RenderBox::hasOverridingLogicalWidthLength const):
(WebCore::RenderBox::clearOverridingLogicalWidthLength):
* Source/WebCore/rendering/RenderBoxModelObject.cpp:
(WebCore::RenderBoxModelObject::continuation const):
(WebCore::RenderBoxModelObject::inlineContinuation const):
(WebCore::RenderBoxModelObject::forRendererAndContinuations):
(WebCore::RenderBoxModelObject::continuationChainNode const):
(WebCore::RenderBoxModelObject::insertIntoContinuationChainAfter):
(WebCore::RenderBoxModelObject::removeFromContinuationChain):
(WebCore::RenderBoxModelObject::ensureContinuationChainNode):
(WebCore::RenderBoxModelObject::firstLetterRemainingText const):
(WebCore::RenderBoxModelObject::setFirstLetterRemainingText):
(WebCore::RenderBoxModelObject::clearFirstLetterRemainingText):
* Source/WebCore/rendering/RenderFlexibleBox.cpp:
(WebCore::RenderFlexibleBox::cachedChildIntrinsicContentLogicalHeight const):
(WebCore::RenderFlexibleBox::setCachedChildIntrinsicContentLogicalHeight):
(WebCore::RenderFlexibleBox::clearCachedChildIntrinsicContentLogicalHeight):
(WebCore::RenderFlexibleBox::cacheChildMainSize):
(WebCore::RenderFlexibleBox::clearCachedMainSizeForChild):
(WebCore::RenderFlexibleBox::computeFlexBaseSizeForChild):
(WebCore::RenderFlexibleBox::maybeCacheChildMainIntrinsicSize):
* Source/WebCore/rendering/RenderFlexibleBox.h:
* Source/WebCore/rendering/RenderFragmentContainer.cpp:
(WebCore::RenderFragmentContainer::renderBoxFragmentInfo const):
(WebCore::RenderFragmentContainer::setRenderBoxFragmentInfo):
(WebCore::RenderFragmentContainer::takeRenderBoxFragmentInfo):
(WebCore::RenderFragmentContainer::removeRenderBoxFragmentInfo):
* Source/WebCore/rendering/RenderFragmentContainer.h:

Canonical link: https://commits.webkit.org/273101@main
